sycl/include/sycl/multi_ptr.hpp

```diff
@@ -552,10 +548,9 @@ class __SYCL_TYPE(multi_ptr) multi_ptr<const void, Space, DecorateAddress> {
   template <typename ElementType,
             typename = typename std::enable_if_t<std::is_const_v<ElementType>>>
   explicit operator multi_ptr<ElementType, Space, DecorateAddress>() const {
-    return multi_ptr<ElementType, Space, DecorateAddress>{
-        detail::cast_AS<typename multi_ptr<ElementType, Space,
+    return {static_cast<typename multi_ptr<ElementType, Space,
```

Comment:
We are casting from a `__decorated__ const void *` to `__decorated__ const something *` here.